packages/puppeteer-crawler/src/internals/utils/puppeteer_utils.ts

```diff
@@ -237,12 +238,33 @@ export async function blockRequests(page: Page, options: BlockRequestsOptions =
 
     const patternsToBlock = [...urlPatterns, ...extraUrlPatterns];
 
-    if (page._client instanceof Function) {
-        await page._client().send('Network.setBlockedURLs', { urls: patternsToBlock });
-    } else {
-        // @ts-expect-error for older puppeteer (<14.4)
-        await page._client.send('Network.setBlockedURLs', { urls: patternsToBlock });
+    // We use CDP commands instead of request interception as the latter disables caching, which is not ideal
+    await sendCDPCommand(page, 'Network.setBlockedURLs', { urls: patternsToBlock });
+}
+
+export async function sendCDPCommand<T extends keyof ProtocolMapping.Commands>(
```

Comment:
not sure if we reexport everything from this file, if so, we should add `@internal` jsdoc here